nms/app/views/equipment/__tests__/FEGEquipmentGatewayTest.tsx

```diff
@@ -356,4 +361,55 @@ describe('<FEGEquipmentGateway />', () => {
       lastFalloverTime,
     );
   });
+
+  it('adds a federation gateway', async () => {
+    const {
+      findByText,
+      findByRole,
+      getByText,
+      getByTestId,
+      getByPlaceholderText,
+    } = render(<Wrapper />);
+
+    mockAPI(MagmaAPI.upgrades, 'networksNetworkIdTiersGet', ['default']);
+    const fegGatewayPostMock = mockAPI(
+      MagmaAPI.federationGateways,
+      'fegNetworkIdGatewaysPost',
+    );
+
+    const description = 'hello I am a federated gateway';
+    const id = 'test_gw2';
+    const hardware_id = '4dfe212f-df33-4cd2-910c-41892a042fee';
+    const name = 'test_gateway2';
+    const tier = 'default';
+
+    fireEvent.click(await findByText('Add Gateway'));
+
+    const addDialog = await findByRole('dialog');
+    expect(addDialog).toBeInTheDocument();
+
+    const setField = (placeholder: string, value: string) =>
+      fireEvent.change(getByPlaceholderText(placeholder), {
+        target: {value},
+      });
+    setField('Gateway 1', name);
+    setField('Sample Gateway description', description);
+    setField('Eg. 4dfe212f-df33-4cd2-910c-41892a042fee', hardware_id);
+    setField('<country>_<org>_<location>_<sitenumber>', id);
+    fireEvent.change(getByTestId('upgradeTier'), {
+      target: {value: tier},
+    });
+
+    fireEvent.click(getByText('Save'));
+
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+    const {gateway} = fegGatewayPostMock.mock
+      .calls[0][0] as FederationGatewaysApiFegNetworkIdGatewaysPostRequest;
+
+    expect(gateway.description).toEqual(description);
```

Comment:
Do we verify that the gateway is added ?

Reply:
Yes and no.
Yes, it was tested that the correct API call was sent against the (orc8r) backend, however,
no, the backend was not mocked or a response from it expected.